RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -49,27 +53,49 @@ class ManageSubscriptionsViewModel: ObservableObject {
     @Published
     private(set) var refundRequestStatusMessage: String?
 
+    var promotionalOffer: PromotionalOffer? {
+        return promotionalOfferViewModel.promotionalOffer
+    }
+
+    var promoOfferDetails: CustomerCenterConfigData.HelpPath.PromotionalOffer? {
+        return promotionalOfferViewModel.promoOfferDetails
+    }
+
+    var localization: CustomerCenterConfigData.Localization? {
+        return promotionalOfferViewModel.localization
+    }
+
+    var product: StoreProduct? {
+        return promotionalOfferViewModel.product
+    }
+
     private var purchasesProvider: ManageSubscriptionsPurchaseType
+    private var promotionalOfferViewModel: PromotionalOfferViewModel
```

Comment:
I thought this was the cleanest, so the `PromotionalOfferViewModel` gathers all logic around promo offers and can be reused from both the `ManageSubscriptionsViewModel` and whatever other place, making it also easier to test

Reply:
Hmm ViewModels depending on other ViewModels could be a source of confusion. Maybe `PromotionalOfferViewModel` can be a UseCase (or 2) instead? 

I see that `PromotionalOfferViewModel` does 2 things right now. Since a UseCase generally only does 1 thing, what do you think of the following split? 
- `PurchasePromoUseCase`, which does what `purchasePromo()` does atm.
- `LoadPromoUseCase`, which does what `loadPromo()` does atm. 

These 2 UseCases can then be reused wherever we want.

Reply:
What about holding the promotional offer data to build the view? That's mainly why I made it a view model, because it holds some data to build the view

Reply:
Maybe the UseCase can return that data instead of setting member fields? Then the `ManageSubscriptionsViewModel` can keep a hold of it how it wants to.

Reply:
yeah, I think that makes sense. I was refactoring the code for another thing and this is making me suffer a bit, so I think I am going to create a use case as you said and make `ManageSubscriptionsViewModel` and `PromotionalOfferViewModel` hold their own data. That's cleaner